### ISSUE TITLE
SyntaxError location for `*args` in class bases is inconsistent

### ISSUE THREAD

**Issue by tusharsadhwani** [python/cpython#93165]
Code snippet:

```python
class C(list[int]): ...
```

On Python 3.9:
```pycon
>>> class C(list[*a]): ...
  File "<stdin>", line 1
    class C(list[*a]): ...
                   ^
SyntaxError: invalid syntax
>>> 
```

On Python 3.10:
```pycon

>>> class C(list[*a]): ...
  File "<stdin>", line 1
    class C(list[*a]): ...
           ^
SyntaxError: expected ':'
>>>
```

I'd have expected it to point directly at the `*` in both cases.

**Comment by AlexWaygood** [comment]
Reproduced on 3.10 (but not on 3.11, where this isn't a `SyntaxError` at all due to PEP 646).

**Comment by wookie184** [comment]
I believe this is a specific case of #92858, which is fixed by #92894

**Comment by tusharsadhwani** [comment]
Correct, closing.
